### DISCUSSION TITLE
VPN Drops out every minute or so. Everything was working fine before I changed VPN providers to Nord.

### DISCUSSION THREAD

**Discussion by amrdeus2** [haugene/docker-transmission-openvpn#2815] [Q&A] [answered]
Hi amazing community!

My transmission container was working great before but I had to change my VPN provider and now I’m having consistent and frequent drop outs with all my other containers that use Transmissions VPN. So. Radarr for instance will open fine, but then 2 minutes later it will disconnect. If I keep refreshing, it will eventually work about 10% of the time, but then drop out again. And all the containers need each other. So for instance, Radarr is pretty useless without Prowlarr. So they need to be running at the same time. Only way to get them working is to restart all the containers. They all work for 2 minutes but then they start dying. 

Running off Mac Pro - OMV - Docker

I use docker compose. The only thing I changed are the environment options:

>     environment:
>       - OPENVPN_PROVIDER=nordvpn
>       - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
>       - OPENVPN_CONFIG=australia
>       - OPENVPN_USERNAME=********
>       - OPENVPN_PASSWORD=*******
>       - LOCAL_NETWORK=192.168.1.0/24
>       - NORDVPN_COUNTRY=sydney
>       - NORDVPN_CATEGORY=legacy_p2p
>       - NORDVPN_PROTOCOL=udp
>       - NORDVPN_SERVER=sg460.nordvpn.com

The last line was added just to see if that helped because it was on the Provider Specific Settings example. I tried with and without that line. I also tried TCP instead of UDP.

Nothing else has changed besides those lines and now its not working well.

Transmission Error I get every couple of minutes:

> “Could not connect to the server. You may need to reload the page to reconnect.”

> Transmission log:
> Transmission will run as
> -------------------------------------
> User name:   root
> User uid:    0
> User gid:    0
> -------------------------------------
> 
> Updating Transmission settings.json with values from env variables
> Attempting to use existing settings.json for Transmission
> Successfully used existing settings.json /config/transmission-home/settings.json
> Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to [REDACTED]
> Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed
> Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete
> Overriding rpc-password because TRANSMISSION_RPC_PASSWORD is set to [REDACTED]
> Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
> Overriding rpc-username because TRANSMISSION_RPC_USERNAME is set to
> Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch
> sed'ing True to true
> STARTING TRANSMISSION
> Transmission startup script complete.
> 2024-03-21 10:21:40 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
> 2024-03-21 10:21:40 Initialization Sequence Completed

Radarr (which uses the Transmission network) Logs:

> [v4.7.5.7809] System.Net.Http.HttpRequestException: Connection refused (localhost:9091)
>  ---> System.Net.Sockets.SocketException (111): Connection refused
>    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)

I'm pretty new to all this. So this could be a basic issue like DNS settings or something I know little about. This was all working fine with a very cheap VPN, so Nord is an upgrade and should be working much better.
Any help would be really appreciated!

**Comment by ilike2burnthing** [comment]
Remove `OPENVPN_CONFIG` and `NORDVPN_SERVER`.

For `NORDVPN_COUNTRY` use the 2 letter country code, e.g. `AU` - https://api.nordvpn.com/v1/servers/countries

Full instructions for NordVPN here - https://haugene.github.io/docker-transmission-openvpn/provider-specific/#nordvpn

**Reply by amrdeus2** [reply]
Thanks very much for trying to help! I gave that a shot and it still connects but then drops out extremely frequently. 

```
      - OPENVPN_PROVIDER=nordvpn
      - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
      - OPENVPN_USERNAME=xxxxxxx
      - OPENVPN_PASSWORD=xxxxxxx
      - LOCAL_NETWORK=192.168.1.0/24
      - NORDVPN_COUNTRY=AU
      - NORDVPN_CATEGORY=legacy_p2p
      - NORDVPN_PROTOCOL=tcp

```
I tried reading the documentation before I posted but I'm quite new to all this, so it doesn't make much sense to me. I even tried with my old "Custom VPN" settings and had the same issue. I tried TCP and UDP as well. 

Any other ideas by chance? Or can you point me in the direction of troubleshooting this?

**Reply by ilike2burnthing** [reply]
Try an OpenVPN (or specific NordVPN) Docker container as a proxy, see if you have the same issue.

**Comment by amrdeus2** [comment]
Thanks very much for that! I was able to get it working fine with this method. But it'd be good to find out why this is happening. Because using another container defeats the purpose of Transmission OpenVPN. Here is my full compose file (with just sonarr and transmission) in case that shows any obvious errors. (Again this worked perfectly with a different VPN provider. All that is different is the NordVPN stuff)

```
> networks:
>   Arr:
>     external: true
> 
> services:
>   transmission-openvpn:
>     container_name: transmission-openvpn
>     cap_add:
>       - NET_ADMIN
>     volumes:
>       - /media/USB/data/apps/transmission/custom/:/etc/openvpn/custom/
>       - /media/USB/data/torrents:/data/torrents
>       - /media/USB/data/watch:/data/watch
>       - /media/USB/data/apps/transmission:/config
>     environment:
>       - OPENVPN_PROVIDER=nordvpn
>       - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
>       - OPENVPN_USERNAME=xxxx
>       - OPENVPN_PASSWORD=xxxx
>       - LOCAL_NETWORK=192.168.1.0/24
>       - NORDVPN_COUNTRY=US
>       - NORDVPN_CATEGORY=legacy_p2p
>       - NORDVPN_PROTOCOL=tcp
> 
>     logging:
>       driver: json-file
>       options:
>         max-size: 10m
>     ports:
>       - '9091:9091'
>       - 8888:8888/tcp # HTTP proxy
>       - '8989:8989'   # port for sonarr
>       - '9696:9696'   # port for prowlarr
>       - '6767:6767'   # port for bazarr
>       - '8686:8686'   # port for lidarr
>       - 8080:8080     # For qbittorrent
>       - 6881:6881     # For qbittorrent
>       - 6881:6881/udp # For qbittorrent
>     restart: always
>     image: haugene/transmission-openvpn
>     networks:
>       - Arr
>     labels:
>       - "deunhealth.restart.on.unhealthy=true"
> 
>   sonarr:
>     image: lscr.io/linuxserver/sonarr:latest
>     container_name: sonarr
>     cap_add:
>       - NET_ADMIN
>     environment:
>       - puid=998
>       - pgid=100
>       - TZ=Australia/Sydney
>     volumes:
>       - /media/USB/data/apps/sonarr:/config
>       - /media/USB/data:/data
>     restart: unless-stopped
>     network_mode: "service:transmission-openvpn"
>     depends_on:
>       - transmission-openvpn
>     networks:
>       - Arr
>     healthcheck:
>       test: curl -f http://localhost:9091 || exit 1
>       interval: 5m00s
>       timeout: 10s
>       retries: 2
>     labels:
>       - "deunhealth.restart.on.unhealthy=true"
```

**Reply by ilike2burnthing** [reply]
The other container was just a troubleshooting step, not a proposed solution. It narrows the issue down to the image or your configuration of the container, rather than your device, network, ISP, etc.. As I'm using NordVPN without issue, then it would seem the issue is either your config (not that I can see one above), or some edge case container issue.

**Reply by ilike2burnthing** [reply]
Does Sonarr also need the `NET_ADMIN` capability?

**Reply by amrdeus2** [reply]
No idea. But I tried without and the exact same thing. Any other ideas?

**Reply by ilike2burnthing** [reply]
You can't use `localhost` for the Sonarr healthcheck URL, you'd need to change it to the Transmission container hostname or internal Docker IP.

You can also get rid of the `custom` volume map, and just use the one `data` volume map (rather than separate `data/torrents` and `data/watch` mappings).

**Reply by amrdeus2** [reply]
I tried it all. None of it worked.

I'm almost certain it has nothing to do with Sonarr (or the other apps linked to it) because it was all working before and I get the connection error in Transmission as well. Its just far more noticible in the other apps because it does connect again quickly but the other apps don't refresh the "depends_on:" as often I'd assume.

Again, it was all working before with a different VPN. Does transmission do any firewall or DNS things?

**Reply by ilike2burnthing** [reply]
Stop all containers, including Transmission. Start just the Transmission container. After 30s or so confirm that only the Tranmission container is running. Do you still have the same issue?

**Reply by amrdeus2** [reply, marked as answer]
Ok that kind of helped! Thanks very much!

Turning other containers off pointed me in the direction of looking at dependencies in other containers. For some reason it looks like my healthcheck was partially deleted. I just added the following and it worked fine:

"    healthcheck:
      test: curl -f http://192.168.x.xxx:9091 || exit 1
      interval: 5m00s
      timeout: 10s
      retries: 2
"
This should've been obvious. I'm sorry for this whole hassle. Thanks so much for your help, honestly wouldn't have done it without you!
